### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Ihmoda**

### CODE REVIEW

File: curriculum/js2/3.js [hunk 1]
```diff
@@ -5,10 +5,31 @@
...
+  if (num % val === 0) {
+    return false;
+  }
+
+  return isPrime(num, val + 1);
 };
 
 module.exports = {
   solution,
+  isPrime,
```
Comment (songz) [inline curriculum/js2/3.js]: no need to export this.

File: curriculum/js2/3.js [hunk 2]
```diff
@@ -5,10 +5,31 @@
...
+  }
+
+  if (!isPrime(a[idx])) {
+    a[idx] = 1;
+  }
+
+  return solution(a, idx + 1);
+};
+
+const isPrime = (num, val = 2) => {
```
Comment (songz) [inline curriculum/js2/3.js]: instead of writing this over again, try importing it. `const {isPrime} = require('../js1/6.js');` Is it problem 6? I forgot.

**Review by seemcat — CHANGES_REQUESTED** [review]

File: curriculum/js2/3.js [hunk 3]
```diff
@@ -5,8 +5,18 @@
  * @returns {array}
  */
```
[the remark this answers is not in the document]
Comment (Ihmoda) [reply curriculum/js2/3.js:7]: This is curriculum/js2/3.js
 * Given an array, return an array of the same length
 *   where all elements that are not prime is changed to 1

**Comment by Ihmoda** [comment]
Original Problem: 

/**
 * Given an array, return an array of the same length
 *   where all elements that are not prime is changed to 1
 * @param {array} a
 * @returns {array}
 */

const solution = (a) => {
  return [];
};

module.exports = {
  solution,
};

File: curriculum/js2/3.js [hunk 4]
```diff
@@ -5,8 +5,18 @@
...
 
-const solution = (a) => {
-  return [];
+let isPrime = require('../js1/7.js').solution;
```
Comment (songz) [inline curriculum/js2/3.js]: always use `const` unless you can't.

File: curriculum/js2/3.js [hunk 5]
```diff
@@ -5,8 +5,18 @@
  * @returns {array}
  */
 
-const solution = (a) => {
-  return [];
+const isPrime = require('../js1/7.js').solution;
+
+const solution = (a, idx = 0) => {
+  if (idx === a.length) {
```
Comment (songz) [inline curriculum/js2/3.js:11]: if your `if` condition only has one line, you don't need `{}`. so your code could be as simple as:
```
const solution = (a, idx = 0) => {
  if(idx === a.length) return a;
  if(!isPrime(a[idx])) a[idx] = 1;
  return solution (a, idx + 1);
}
```

**Review by songz — APPROVED** [review]

**Pull request merged**